### PR TITLE
Talisker format node logging

### PR DESCRIPTION
**Pull request by earnubs**
logs

Replace util.log with Winston configured to output talisker format logs from named logger instances, or the default winston logger.

Log levels have been set to only 'info' and 'debug' (see https://dave.cheney.net/2015/11/05/lets-talk-about-logging). 

Also, express request logging is via winston too, and uses the same talisker format.

Example:
```
2016-11-17 17:19:32.387Z INFO session "Starting with development default session secret"
2016-11-17 17:19:32.394Z INFO express "Express server listening on http://127.0.0.1:3000"
2016-11-17 17:19:33.914Z INFO express "HTTP GET /" statusCode=302, url=/, host=localhost:3000, connection=keep-alive, cache-control=max-age=0, upgrade-insecure-requests=1, user-agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.98 Safari/537.36", accept=text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8, dnt=1, accept-encoding="gzip, deflate, sdch, br", accept-language=en-GB,en;q=0.8,en-US;q=0.6,fr;q=0.4,fr-FR;q=0.2, method=GET, httpVersion=1.1, originalUrl=/, , responseTime=36
```

To use:

```
import logging from './logging';
const logger = logging.getLogger('myLogger'); // 'myLogger' is the id of this logger instance, and will appear in the log output after log level
logger.info('hello');
```

Will produce:

```
2016-11-08 21:16:16.873Z INFO myLogger "hello"
```

For future use, this should be published as it's own module (along with some other talisker features) `node-talisker`.

### CODE REVIEW

File: webpack/build-config.js [hunk 1]
```diff
@@ -42,5 +36,10 @@ module.exports = {
   },
   postcss: function () {
     return [ vars({ variables: () => sharedVars }), autoprefixer ];
+  },
```
Comment (earnubs) [inline webpack/build-config.js]: do this in dev config too, rather than conf arg to dev server

File: src/server/server.js [hunk 2]
```diff
@@ -1,44 +1,43 @@
...
-app.use(session(sessionConfig(conf)));
-
+// config
 if (app.get('env') === 'production') {
   app.set('trust proxy', 1);
 }
-
+// FIXME sstewart 07-Nov-2016 simplify config to host and port
+app.locals.host = conf.get('SERVER:HOST') || appUrl.hostname;
+app.locals.port = conf.get('SERVER:PORT') || appUrl.port;
```
Comment (earnubs) [inline src/server/server.js]: This ... isn't related to logging. I just wanted to tidy it up.

File: webpack/webpack-dev-server.js [hunk 3]
```diff
@@ -33,17 +34,15 @@ webpackMiddleware.waitUntilValid(() => {
 });
 
 app.use(webpackMiddleware);
-
 app.use(webpackHotMiddleware(compiler));
```
Comment (earnubs) [inline webpack/webpack-dev-server.js]: we can actually configure these to use our logger, but it's dev only so I didn't add it to the PR https://gist.github.com/earnubs/b0917bebda129525ce2bd70d1ad39ba0

File: src/server/server.js [hunk 4]
```diff
@@ -1,43 +1,51 @@
...
+app.use(helmet());
+app.use(session(sessionConfig(conf)));
 app.use(Express.static(__dirname + '/../public', { maxAge: '365d' }));
 
+// routes
 app.use('/', routes.login);
 app.use('/api', routes.api);
 app.use('/', routes.universal);
 
-app.use(raven.middleware.express.errorHandler(conf.get('SENTRY_DSN')));
```
Comment (bartaz) [inline src/server/server.js]: This was removed and exchanged for winston `errorLogger` - we don't want express errors to be sent to sentry?

Comment (bartaz) [inline src/server/server.js]: According to docs both are needed. `requestHandler` first and `errorHandler` - before any other error middleware.

File: src/server/server.js [hunk 5]
```diff
@@ -1,45 +1,53 @@
...
 app.use(Express.static(__dirname + '/../public', { maxAge: '365d' }));
 
+// routes
 app.use('/', routes.login);
 app.use('/api', routes.api);
 app.use('/', routes.universal);
 
+// FIXME sstewart 18-Nov-16 won't ever log because of
+// https://github.com/canonical-ols/javan-rhino/issues/210
 app.use(raven.middleware.express.errorHandler(conf.get('SENTRY_DSN')));
```
Comment (earnubs) [inline src/server/server.js]: @bartaz sorry, back now :)

**Pull request merged**